protobufs/livekit_rtc_v2.proto

```diff
@@ -0,0 +1,156 @@
+// Copyright 2023 LiveKit, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package livekit;
+option go_package = "github.com/livekit/protocol/livekit";
+option csharp_namespace = "LiveKit.Proto";
+option ruby_package = "LiveKit::Proto";
+
+import "livekit_rtc.proto";
+import "livekit_models.proto";
+
+message Signalv2ClientMessage {
+  // monotonically increasing number starting at some non-zero number (should not rollover)
+  uint32 message_id = 1;
+
+  // last remote message processed
+  uint32 last_processed_remote_message_id = 2;
+
+  oneof message {
+    ConnectRequest connect_request = 3;
+    ReconnectRequest reconnect_request = 4;
+    AudioTrack audio_track = 5;
+    VideoTrack video_track = 6;
+    SessionDescription publisher_sdp = 7;
+    SessionDescription subscriber_sdp = 8;
+    TrickleRequest ice_candidate = 9;
+    MuteTrackRequest mute_track_request = 10;
+    UpdateSubscription update_subscription = 11;
+    UpdateTrackSettings update_track_settings = 12;
+    LeaveRequest leave_request = 13;
+    SubscriptionPermission update_subscription_permission = 14;
+    SimulateScenario simulate_scenario = 15;
+    UpdateParticipantMetadata update_participant_metadata = 16;
+    Ping ping = 17;
+    UpdateLocalAudioTrack update_local_audio_track = 18;
+    UpdateLocalVideoTrack update_local_video_track = 19;
+  }
+}
+
+message Signalv2ClientEnvelope {
+  // collection of client -> server messages
+  repeated Signalv2ClientMessage client_messages = 1;
+}
```

Comment:
Making an envelope type to send a collection of messages. Useful for replaying messages from a certain point in time.